packages/cdk-assets/test/private/docker.test.ts

```diff
@@ -91,4 +97,164 @@ describe('Docker', () => {
       expect(imageExists).toBe(false);
     });
   });
+
+  describe('dockerConfigFile', () => {
+    let docker: Docker;
+
+    beforeEach(() => {
+      docker = new Docker();
+      process.env = { ..._ENV };
+    });
+
+    afterEach(() => {
+      process.env = _ENV;
+    });
+
+    test('Can be overridden by CDK_DOCKER_CONFIG_FILE', () => {
+      const configFile = '/tmp/insertfilenamehere_docker_config.json';
+      process.env.CDK_DOCKER_CONFIG_FILE = configFile;
+
+      expect(docker.dockerConfigFile()).toEqual(configFile);
+    });
+
+    test('Uses homedir if no process env is set', () => {
+      expect(docker.dockerConfigFile()).toEqual(path.join(os.userInfo().homedir, '.docker', 'config.json'));
+    });
+  });
+
+  describe('dockerConfig', () => {
+    let docker: Docker;
+    const configFile = '/tmp/foo/bar/does/not/exist/config.json';
+    afterEach(() => {
+      jest.resetModules();
+      jest.resetAllMocks();
+    });
+
+    beforeEach(() => {
+      docker = new Docker();
+      process.env.CDK_DOCKER_CONFIG_FILE = configFile;
+    });
+
+    afterEach(() => {
+      mockfs.restore();
+      process.env = { ..._ENV };
+    });
+
+    test('returns empty object if no config exists', () => {
+      expect(docker.dockerConfig()).toEqual({});
+    });
+
+    test('returns parsed config if it exists', () => {
+      mockfs({
+        [configFile]: JSON.stringify({
+          proxies: {
+            default: {
+              httpProxy: 'http://proxy.com',
+              httpsProxy: 'https://proxy.com',
+              noProxy: '.amazonaws.com',
+            },
+          },
+        }),
+      });
+
+      const config = docker.dockerConfig();
+      expect(config).toBeDefined();
+      expect(config.proxies).toEqual({
+        default: {
+          httpProxy: 'http://proxy.com',
+          httpsProxy: 'https://proxy.com',
+          noProxy: '.amazonaws.com',
+        },
+      });
+    });
+  });
+
+  describe('configureCdkCredentials', () => {
+    let docker: Docker;
+    const credsFile = '/tmp/foo/bar/does/not/exist/config-cred.json';
+    const configFile = '/tmp/foo/bar/does/not/exist/config.json';
+
+    afterEach(() => {
+      jest.resetModules();
+      jest.resetAllMocks();
+    });
+
+    beforeEach(() => {
+      docker = new Docker();
+      process.env.CDK_DOCKER_CREDS_FILE = credsFile;
+      process.env.CDK_DOCKER_CONFIG_FILE = configFile;
+    });
+
+    afterEach(() => {
+      mockfs.restore();
+      process.env = { ..._ENV };
+    });
+
+    test('returns false if no cred config exists', () => {
+      expect(docker.configureCdkCredentials()).toBeFalsy();
+    });
```

Comment:
Was this expected behavior before, or is this new?

If not new, would this have been covered by one of the above tests - i.e. would this have resulted in a shell error?